libraries/c_sdk/aws/defender/src/aws_iot_defender_api.c

```diff
@@ -413,7 +420,7 @@ static void _metricsPublishRoutine( IotTaskPool_t pTaskPool,
 
         IotTaskPool_ScheduleDeferred( IOT_SYSTEM_TASKPOOL,
                                       _disconnectJob,
-                                      _defenderToMilliseconds( AWS_IOT_DEFENDER_WAIT_SERVER_MAX_SECONDS ) );
+                                      0 );
```

Comment:
We are already checking for whether the report is sent or not with QoS1. So when hitting this line, can simply disconnect without waiting.

Reply:
Yeah, report being sent is ok. But we want to get response from defender service to see if the report is accepted.

https://github.com/aws/amazon-freertos/blob/750bbdf0e1ad9ad53a6dc976ed80541955bfe094/libraries/c_sdk/aws/defender/src/aws_iot_defender_api.c#L511

Reply:
I see. 

Something I would like to understand is that, is it in defender requirement each publish should be synchronous? In other words, why not having two threads one for report publish, the other for handling defender service callback?

Reply:
synchronous publish is not required. I think it is OK to make it async.

Reply:
Talked offline. The other consideration in original implementation is to **try to** wait for server response before disconnect. (In a sense that, if user does care about ```/rejected``` or ```/accepted``` response, the user needs to figure out a reasonable ```AWS_IOT_DEFENDER_WAIT_SERVER_MAX_SECONDS```. Depending on which region the device is trying to connect to from where, this value could be different.)

Since we do have a plan for updating the defender report scheduling, I'm going to revert the change to use macro here. But the macro value will be updated to zero. The follow-up items would be:
- update scheduling.
- update demo, to actually show how callback works.